presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java

```diff
@@ -74,9 +89,29 @@ protected Void visitLogicalBinaryExpression(LogicalBinaryExpression node, Materi
         if (!node.getOperator().equals(LogicalBinaryExpression.Operator.AND)) {
             throw new SemanticException(NOT_SUPPORTED, node, "Only AND operator is supported for join criteria for materialized view.");
         }
+
+        visitLogicalBinaryExpressionChild(node.getLeft(), context);
+        visitLogicalBinaryExpressionChild(node.getRight(), context);
+
         return super.visitLogicalBinaryExpression(node, context);
     }
 
+    protected void visitLogicalBinaryExpressionChild(Expression node, MaterializedViewPlanValidatorContext context)
+    {
+        if (node == null) {
+            return;
+        }
+
+        if (node instanceof ComparisonExpression) {
+            if (node instanceof LogicalBinaryExpression) {
+                visitLogicalBinaryExpression((LogicalBinaryExpression) node, context);
+            }
+            else {
+                throw new SemanticException(NOT_SUPPORTED, node, "Only comparison or logical binary expressions are supported for join criteria for materialized view.");
+            }
+        }
+    }
+
```

Comment:
Pls check on the body of ComparisonExpression.

A better way is to implement the visitComparisonExpression() method, ex.,

```suggestion
    @Override
    protected Void visitComparisonExpression(ComparisonExpression node, MaterializedViewPlanValidatorContext context)
    {
        if (!context.isProcessingJoinNode()) {
            return super.visitComparisonExpression(node, context);
        }

        if (!node.getOperator().equals(ComparisonExpression.Operator.EQUAL)) {
            throw new SemanticException(NOT_SUPPORTED, node, "Only EQUAL join is supported for materialized view.");
        }
        return super.visitComparisonExpression(node, context);
   }

```